### PR TITLE
Patch library

### PR DESCRIPTION
**Pull request by stephenkr**
This PR will:
1. Update master to v0.1.8 (Ive already force pushed the tree to this version)
2. Patch the library to stop leaking to the global scope
3. Update package.json to be compatible with gilt components
4. Minor clean up

### CODE REVIEW

File: messageformat.js [hunk 1]
```diff
@@ -1554,28 +1554,15 @@
...
+      this.precompile(
+        this.parse( message ),
+        methods
+      )
+    ))(MessageFormat, methods);
   };
 
+  exports.MessageFormat = MessageFormat;
 
-  if (typeof exports !== 'undefined') {
```
Comment (Meesayen) [inline messageformat.js]: keep the CJS part, or we won't be able to use it without the gilt.define trickery

Comment (stephenkr) [inline messageformat.js]: Sure, i'll add it back in.

File: messageformat.js [hunk 2]
```diff
@@ -1554,28 +1554,26 @@
...
-        )
-    ))(MessageFormat);
+      this.precompile(
+        this.parse( message ),
+        methods
+      )
+    ))(MessageFormat, methods);
   };
 
+  if (typeof define === 'function' && define.amd) {
```
Comment (Meesayen) [inline messageformat.js]: this will never be our case, see line 4

Comment (stephenkr) [inline messageformat.js]: True, i'll remove it.

**Comment by stephenkr** [comment]
Closed in favor of current implementation of third party dependencies.

https://gerrit.gilt.com/#/c/78277/

**Pull request closed without merging**